server/controllers/lostBedsController.ts

```diff
@@ -0,0 +1,52 @@
+import type { Response, Request, RequestHandler } from 'express'
+
+import type { LostBedDto } from 'approved-premises'
+import LostBedService from '../services/lostBedService'
+import { catchValidationErrorOrPropogate, fetchErrorsAndUserInput } from '../utils/validation'
+import { convertDateAndTimeInputsToIsoString } from '../utils/utils'
+
+export default class LostBedsController {
+  constructor(private readonly lostBedService: LostBedService) {}
+
+  new(): RequestHandler {
+    return async (req: Request, res: Response) => {
+      const { premisesId } = req.params
+      const { errors, errorSummary, userInput } = fetchErrorsAndUserInput(req)
+
+      const lostBedReasons = await this.lostBedService.getReferenceData(req.user.token)
+
+      res.render('lostBeds/new', {
+        premisesId,
+        lostBedReasons,
+        errors,
+        errorSummary,
+        ...userInput,
+      })
+    }
+  }
+
+  create(): RequestHandler {
+    return async (req: Request, res: Response) => {
+      const { premisesId } = req.params
+
+      const { startDate } = convertDateAndTimeInputsToIsoString(req.body, 'startDate')
+      const { endDate } = convertDateAndTimeInputsToIsoString(req.body, 'endDate')
+
+      const lostBed: LostBedDto = { ...req.body.lostBed, startDate, endDate }
+
+      try {
+        const { id } = await this.lostBedService.createLostBed(req.user.token, premisesId, lostBed)
+        res.redirect(`/premises/${premisesId}/lostBeds/${id}/confirmation`)
+      } catch (err) {
+        catchValidationErrorOrPropogate(req, res, err, `/premises/${premisesId}/lostBeds/new`)
+      }
+    }
+  }
+
+  confirm(): RequestHandler {
```

Comment:
Do we need a confirmation screen here? Or should we stick with the flash message pattern we've used for the smaller journeys?

Reply:
Yeah flash probably makes sense here